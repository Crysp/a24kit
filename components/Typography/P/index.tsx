import React from 'react';
import Wrapper from './styled';

export interface Props {
    size: 1 | 2 | 3
    light: boolean
    children: React.ReactNode
}

/**
 * General component description in JSDoc format. Markdown is *supported*.
 * @develop
 */
export default class P extends React.PureComponent<Props> {
    static defaultProps = {
        size: 1,
        light: false,
    };

    render() {
        const { children, ...other } = this.props;
        return <Wrapper {...other}>{children}</Wrapper>;
    }
}
