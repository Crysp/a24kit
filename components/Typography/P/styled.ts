import styled, { css } from 'styled-components';

const getSize = (props) => {
    if (props.size === 2) {
        return css`
            font-size: 13px;
            line-height: 20px;
            font-weight: 400;
        `;
    }
    if (props.size === 3) {
        return css`
            font-size: 12px;
            line-height: 20px;
            font-weight: 400;
        `;
    }

    return css`
        font-size: 14px;
        line-height: 24px;
        font-weight: 400;
    `;
};

interface Props {
    light: boolean
}

export default styled.p<Props>`
    margin: 0;
    font-family: "Open Sans", sans-serif;
    word-wrap: break-word;
    color: ${props => (props.light ? '#9c9c9c' : 'inherit')};
    ${getSize}
    
    & + & {
        margin-top: 12px;
    }
`;
