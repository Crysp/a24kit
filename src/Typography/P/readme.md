```js
<div>
    <P>Параграф первого уровня</P>
    <P size={2}>Параграф второго уровня</P>
    <P size={3}>Параграф третьего уровня</P>
</div>
```