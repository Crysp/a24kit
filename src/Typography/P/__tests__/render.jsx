import React from 'react';
import renderer from 'react-test-renderer';
import P from '../index';


describe('Верстка компонента <P>', () => {
    it('Дефолтная', () => {
        const tree = renderer.create(<P>Paragraph</P>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Первого уровня', () => {
        const tree = renderer.create(<P size={1}>Paragraph</P>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Второго уровня', () => {
        const tree = renderer.create(<P size={2}>Paragraph</P>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Третьего уровня', () => {
        const tree = renderer.create(<P size={3}>Paragraph</P>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
