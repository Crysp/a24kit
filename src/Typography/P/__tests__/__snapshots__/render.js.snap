// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Верстка компонента <P> Второго уровня 1`] = `
<p
  className="p p--size2"
>
  Paragraph
</p>
`;

exports[`Верстка компонента <P> Дефолтная 1`] = `
<p
  className="p p--size1"
>
  Paragraph
</p>
`;

exports[`Верстка компонента <P> Первого уровня 1`] = `
<p
  className="p p--size1"
>
  Paragraph
</p>
`;

exports[`Верстка компонента <P> Третьего уровня 1`] = `
<p
  className="p p--size3"
>
  Paragraph
</p>
`;
